docs/src/data/amplifyMock/services.ts

```diff
@@ -8,7 +10,28 @@ const verifiedUsers = new Map();
 let signUpPassword = '';
 
 const checkFixedCredentials = (username, password) => {
-  return username === FIXED_USERNAME && password === FIXED_PASSWORD;
+  let user = '';
+  if (username.length === 14) {
+    if (username.substring(4) === FIXED_PHONE) {
+      user = username.substring(4);
+    }
+  } else if (username.length === 13) {
+    if (username.substring(3) === FIXED_PHONE) {
+      user = username.substring(3);
+    }
+  } else if (username.length === 12) {
+    if (username.substring(2) === FIXED_PHONE) {
+      user = username.substring(2);
+    }
+  } else if (username.length === 10) {
+    user = username;
+  }
+  return (
+    (username === FIXED_USERNAME ||
+      username === FIXED_EMAIL ||
+      user === FIXED_PHONE) &&
+    password === FIXED_PASSWORD
+  );
```

Comment:
I'd definitiely split this up, this is too complicated and will be hard to maintain later.

What I'd suggest is to break up into three cases: `isUsername`, `isPhone`, `isEmail`, and handle them separately.

```ts
if (isUsername(username)) {
  // some utility function here...
} else if (isPhone(username)) {
  // some utility function here... 
} else if (isEmail(username)) {
  // some utility function here...
 } else {
   // throw some error...
}

Reply:
Talked async, came up with simpler solution